include/deal.II/lac/precondition.h

```diff
@@ -1903,7 +1903,6 @@ namespace internal
       (void)matrix;
       (void)preconditioner;
       AssertThrow(preconditioner.get() != NULL, ExcNotInitialized());
-      AssertDimension(preconditioner->m(), matrix.m());
```

Comment:
Trilinos preconditioners do not define an `m()` operation. Since we are in user land with an arbitrary preconditioner type, the user will need to make sure that the dimension fits. In case the classes come from deal.II, the `vmult()` function will ensure the compatibility of the vector with both the matrix and the preconditioner, so this is fine.